firmware.sh

```diff
@@ -1,19 +1,25 @@
 #!/bin/bash
 
-while getopts s:t:f: flag
+while getopts s:t:f:b: flag
 do
     case "${flag}" in
         s) switch=${OPTARG};;
 		t) ftype=${OPTARG};;
 		f) flash=${OPTARG};;
+		b) beta=${OPTARG};;
     esac
 done
 # Define repository URLs
 CARTOGRAPHER_KLIPPER_REPO="https://github.com/Cartographer3D/cartographer-klipper.git"
 KATAPULT_REPO="https://github.com/Arksine/katapult.git"
 
-# Define local paths where the repositories should be cloned
-CARTOGRAPHER_KLIPPER_DIR="$HOME/cartographer-klipper"
+if [[ $beta == "beta" ]]; then
+	TARBALL_URL="https://api.github.com/repos/Cartographer3D/cartographer-klipper/tarball/master"
+else
+	TARBALL_URL="https://api.github.com/repos/Cartographer3D/cartographer-klipper/tarball/beta-firmware"
+fi
+TEMP_DIR="/tmp/cartographer-klipper"
```

Comment:
Shouldn't this use `mktemp -d` instead?
https://www.man7.org/linux/man-pages/man1/mktemp.1.html